src/amrex/extensions/MultiFab.py

```diff
@@ -132,15 +172,14 @@ def mf_to_xp(self, copy=False, order="F"):
     Returns
     -------
     list of xp.array
-        A list of NumPy or CuPy n-dimensional arrays, for each local block in the
-        MultiFab.
+        A list of NumPy, CuPy or dpnp n-dimensional arrays, for each local block
+        in the MultiFab.
     """
-    import inspect
+    views = []
+    for mfi in self:
+        views.append(self.array(mfi).to_xp(copy, order))
 
-    amr = inspect.getmodule(self)
-    return (
-        self.to_cupy(copy, order) if amr.Config.have_gpu else self.to_numpy(copy, order)
-    )
+    return views
```

Comment:
By changing the `mf_to_xp` function to directly call `array4_to_xp` (instead of doing the same configuration check as in `array4_to_xp` and then calling individual `mf_to_numpy/cupy/dpnp`) we no longer need the `mf_to_cupy` or `mf_to_dpnp` functions. I imagine we would want to keep `mf_to_numpy` in case someone has a specific need to return a numpy array during GPU execution, but I don't see the need to have the `cupy` or `dpnp` versions. @ax3l do you agree?

Reply:
On 2nd thought though, the docs mention the explicit functions and might need a rewrite when we remove them.

But always using to_xp is the right way to write portable code...